src/parsers/utils.hpp

```diff
@@ -52,6 +80,10 @@ namespace se3
     // else return an empty string that will provoke an error in loadPolyhedronFromResource()
     namespace bf = boost::filesystem;
 
+    // In case the mesh_path is such as "file://path/to/mesh", check for /path/to/mesh and return it if it exists
+    if ( bf::exists( bf::path(urdf_mesh_path.substr(6, urdf_mesh_path.size())) ))
+      return std::string(urdf_mesh_path.substr(6, urdf_mesh_path.size()));
```

Comment:
I do not have such examples. The possible errors are just example and I'm sure I can find even more tricky issues but it's not the point. The current code is just not behaving properly because it does not fit with the file naming convention. I'm 99% sure that this code will one day be the cause of some unexpected errors...

Using regex should be a lot more simple and robust. Here is an example to be compile with `-lboost_regex` flag:
```
#include <iostream>
#include <boost/regex.hpp>

using namespace boost;

void run(const std::string str) {
  regex rx ("^(?'scheme'[[:word:]]+://)?(?'path'.+)$");
  cmatch what;
  std::cout << "\nResult for " << str << '\n';
  if(regex_match(str.c_str(), what, rx))
  {
    std::cout << "scheme = " << what["scheme"] << std::endl;
    std::cout << "path = " << what["path"] << std::endl;
  }
}

int main () {
  run("file://path/to/mesh1");
  run("package://robot/path/to/mesh1");
  run("aRepoName/robot/path/to/mesh1");
  run("file:///path/to/mesh1");
  run("package:///path/to/mesh1");
}
```